src/transformStateWithClones.js

```diff
@@ -7,7 +7,81 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  for (const part of actions) {
+    const actionValues = Object.values(part);
+    const actionKeys = Object.keys(part);
+
+    for (let n = 0; n < actionKeys.length; n++) {
+      switch (actionKeys[n]) {
+        case 'type':
+          switch (actionValues[n]) {
+            case 'addProperties':
+              if (result.length === 0) {
+                result.push(Object.assign({}, state, actionValues[n + 1]));
```

Comment:
prev commens was about ude for of loop, also your algorithm is very difficult 
why do you need nested switches?
you can just create ,cope of state, and operate with it 
f.e.:

const copeState = { ...state };

for (const action of  actions) { 
  switch (action.type) {
    case 'addProperties':
        Object.assign(copyState, action.extraData);
        break;
        ......
        that's all what you need for addProperties case, also you can simplify all your cases